azure/scope/managedcontrolplane.go

```diff
@@ -289,6 +296,19 @@ func (s *ManagedControlPlaneScope) VNetSpec() azure.ResourceSpecGetter {
 	}
 }
 
+// AzureFleetsMemberSpec returns the fleet spec.
+func (s *ManagedControlPlaneScope) AzureFleetsMemberSpec() azure.ASOResourceSpecGetter[*asocontainerservicev1preview.FleetsMember] {
+	return &fleetsmember.AzureFleetsMemberSpec{
+		Name:                 s.ClusterName(),
```

Comment:
2 aks clusters can have the same name, if they are in different subscription, or different resource groups.
so this can conflict and the user needs to specify the name.

Reply:
Ah that makes sense. That answers my question from above. Will add a separate field for the member name!

Reply:
shouldn't there be an if condition here? to accept the user's input if it's there?
how do the defaults work?

is there unit tests around the defaulting logic?

Reply:
The webhook defaulting happens in `azuremanagedcontrolplane_default.go` and is tested here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azuremanagedcontrolplane_webhook_test.go

Gonna add a unit test for it on the next change.